app/core/ts/utils/ComponentUtils.ts

```diff
@@ -307,6 +310,22 @@ export function CircuitBoundingBox(all: CullableObject[]): BoundingBox {
     return new BoundingBox(min, max);
 }
 
+// Zooms and translates camera to fit objs with adjustable padding ratio
+export function FitCamera(camera: Camera, objs: CullableObject[], padding: number): void {
+    const bbox = objs.length === 0
+        ? new BoundingBox(EMPTY_CIRCUIT_MIN, EMPTY_CIRCUIT_MAX)
+        : CircuitBoundingBox(objs);
+    camera.setPos(bbox.getCenter());
+
+    const screenSize = camera.getCenter().scale(2); // Bottom right corner of screen
+    const worldSize = camera.getWorldPos(screenSize).sub(camera.getWorldPos(V(0,0))); // World size of camera view
+
+    // Determine which bbox dimension will limit zoom level
+    const xRatio = bbox.getWidth()/worldSize.x;
+    const yRatio = bbox.getHeight()/worldSize.y;
+    camera.zoomBy(Math.max(xRatio, yRatio) * padding);
+}
+
```

Comment:
See above comment.

For 2) see [this](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple)